flink-runtime/src/main/java/org/apache/flink/runtime/operators/GroupReduceCombineDriver.java

```diff
@@ -170,7 +171,14 @@ public void run() throws Exception {
 
 			// write the value again
 			if (!this.sorter.write(value)) {
-				throw new IOException("Cannot write record to fresh sort buffer. Record too large.");
+				if (oversizedRecordCount == Long.MAX_VALUE) {
+					LOG.debug("Number of oversized record has exceeded MAX Long");
+				} else {
+					++oversizedRecordCount;
+					LOG.debug("Cannot write record to fresh sort buffer. Record too large. Oversized record count: {}", oversizedRecordCount);
+				}
+				// simply forward the record
+				this.output.collect((OUT)value);
```

Comment:
I tried to call

this.output.collect(value)
I couldn't figure out how to convert this single value from type IN to type OUT besides direct casting. Could you guys please help me? @StephanEwen @rmetzger